<launch>
	
  <arg name="master_name" default="phantom"/>
  <arg name="publish_rate" default="1000.0"/>
  <arg name="ip_phantom" default="192.168.0.108"/>
  <arg name="ip_system" default="192.168.0.108"/>
  <arg name="write_port_pose" default="34000"/>
  <arg name="write_port_button" default="34001"/>
  <arg name="read_port" default="33000"/>
  <arg name="frame_id" default="world"/>
  
<!-- Phantom node -->
  <node name="omni_state2" pkg="omni_common" type="omni_state2" output="screen"> 
    <param name="omni_name" type="string" value="$(arg master_name)"/>
    <param name="publish_rate" type="double" value="$(arg publish_rate)"/>
  </node>

<!-- Phantom UDP -->
  <node name="phantom_udp_send" pkg="omni_common" type="send_udp_info_phantom.py" output="screen">
	<param name="omni_name" type="string" value="$(arg master_name)"/>
	<param name="write_ip" type="string" value="$(arg ip_system)"/>
    <param name="write_port_pose" type="int" value="$(arg write_port_pose)"/>
    <param name="write_port_button" type="int" value="$(arg write_port_button)"/>
    <param name="publish_rate" type="double" value="$(arg publish_rate)"/>
  </node>
  
  <node name="phantom_udp_receive" pkg="omni_common" type="receive_udp_info_phantom.py" output="screen">
	<param name="omni_name" type="string" value="$(arg master_name)"/>
	<param name="read_ip" type="string" value="$(arg ip_phantom)"/>
    <param name="read_port" type="int" value="$(arg read_port)"/>
    <param name="publish_rate" type="double" value="$(arg publish_rate)"/>
    <param name="frame_id" type="string" value="$(arg frame_id)"/>
  </node>
  
</launch>
